react/features/welcome/route.js

```diff
@@ -1,11 +1,31 @@
+/* global APP */
 import { RouteRegistry } from '../base/navigator';
-
+import { generateRoomWithoutSeparator } from '../base/util';
 import { WelcomePage } from './components';
 
+
+/**
+ * Function that checks if welcome page is enabled and if it isn't
+ * redirects to randomly created conference.
+ *
+ * @param {Object} nextState - Next router state.
+ * @param {Function} replace - Function to redirect to another path.
+ * @returns {void}
+ */
+function onEnter(nextState, replace) {
+    if (!APP.settings.isWelcomePageEnabled()) {
```

Comment:
Both react/features/welcome/route.js and react/features/conference/route.js are supposed to be platform-agnostic (as the simple .js file extension suggests). You put APP and/or imported from modules/ in them which cannot possibly run on mobile.